### PR TITLE
Cores slides

### PR DESCRIPTION
**Pull request by MMysore2**
Current State:

- [ ] Cores slides
    - [x] Add relevant material from old slides
    - [ ] Add tutorial content for slides (Finished 1/2 tutorials)
    - [ ] Fix bugs, typos, etc.

### CODE REVIEW

File: slides/02-Using-gem5/04-cores.md [hunk 1]
```diff
@@ -13,9 +13,643 @@ title: Modeling CPU cores in gem5
...
+timing-small-cache/stats.txt:simOps                                        33954560
+```
+
+---
+## Look at the Number of Execution Cycles
+
+Run the following command
+
+```sh
+grep -ri "numCycles" atomic-normal-cache atomic-small-cache timing-normal-cache timing-small-cache | grep "cores0"
```
Comment (powerjg) [inline slides/02-Using-gem5/04-cores.md:424]: Can we improve this line? Maybe something like `grep -ri cores0.*numCycles`?

File: materials/developing-gem5-models/04-cores/cores-complex.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    num_channels=2,
+    interleaving_size=128,
+    size="1 GiB",
+)
+
+# Use big processor by default
+# processor = big()
+
+# Uncomment for step 3
+processor = LITTLE()
```
Comment (powerjg) [inline materials/developing-gem5-models/04-cores/cores-complex.py:50]: Can you update this so there is a single command line parameter (i.e., using argparse) to select either "big" or "little"

File: materials/developing-gem5-models/04-cores/components/processors.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            + 4 * self._width
+            + 2 * self._rob_size
+            + self._num_int_regs
+            + self._num_fp_regs
+        )
+        return score
+
+# UPDATE FOR STEP 1
+# Configure with width=10, rob_size=40, num_int_regs=50, num_fp_regs=50
+class big(O3CPU):
```
Comment (powerjg) [inline materials/developing-gem5-models/04-cores/components/processors.py:134]: These class names should follow the gem5 style guide. I.e., `Big`

File: materials/developing-gem5-models/04-cores/components/processors.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        super().__init__(
+            width=0,
+            rob_size=0,
+            num_int_regs=0,
+            num_fp_regs=0,
+        )
+
+# UPDATE FOR STEP 1
+# Configure with width=2, rob_size=30, num_int_regs=40, num_fp_regs=40
+class LITTLE(O3CPU):
```
Comment (powerjg) [inline materials/developing-gem5-models/04-cores/components/processors.py:145]: These class names should follow the gem5 style guide. I.e., `Little`

File: slides/02-Using-gem5/04-cores.md [hunk 5]
```diff
@@ -13,9 +13,643 @@ title: Modeling CPU cores in gem5
...
+- Some timing
+- Caches, limited BP
+
+### **DerivO3CPU and MinorCPU**
+
+- Slow
+- Timing
+- Caches, BP
+
+![bg width:1200px Diagram to show different CPU Models](04-cores-imgs/Summary-of-gem5-models-bg-2.png)
```
Comment (powerjg) [inline slides/02-Using-gem5/04-cores.md:190]: These pictures are *very pretty*. Could you make them svgs, and editable with draw.io? The goal with these slides is to be easy to update in the future :smile:

**Review by powerjg — COMMENTED** [review]
I've left some comments below. I also made some direct changes. I rebased on main, so you won't be able to pull cleanly. Assuming you haven't made any changes you want to keep locally you can use the following to check out this branch:
```sh
git checkout cores-slides
git reset --hard origin/cores-slides
```

Thanks for the hard work!

**Comment by powerjg** [comment]
Hopefully this was ready to go 😆

File: materials/03-Developing-gem5-models/05-modeling-cores/completed/processors.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
+# Copyright (c) 2022 The Regents of the University of California
```
Comment (BobbyRBruce) [inline materials/03-Developing-gem5-models/05-modeling-cores/completed/processors.py:1]: You've put these material files in the wrong directory. These should be in in "02-Using-gem5". This is "03-Developing-gem5-models". At present, on the main branch, these files are mixed in with those i use in my "adding an instruction" example.

**Pull request merged**